rotkehlchen/rotkehlchen.py

```diff
@@ -467,6 +467,14 @@ def unlock_user(
         self.user_is_logged_in = True
         log.debug('User unlocking complete')
 
+    def change_password(self, new_password: str) -> bool:
```

Comment:
This is not a good approach. We have the password stored in 3 classes and now we have 3 different one liner functions to update it. Each one calling the other.

Since you are touching this, improve it.

Study the code and see where it should be used and where it can go away.

```
rotkehlchen/rotkehlchen.py:267:13:        self.password = password
rotkehlchen/rotkehlchen.py:489:13:        self.password = ''
rotkehlchen/premium/sync.py:48:13:        self.password = password
rotkehlchen/premium/sync.py:119:53:            self.data.decompress_and_decrypt_db(self.password, result['data'])
rotkehlchen/premium/sync.py:156:80:            b64_encoded_data, our_hash = self.data.compress_and_encrypt_db(self.password)
rotkehlchen/data_handler.py:38:13:        self.password = ''
rotkehlchen/data_handler.py:45:17:            self.password = ''
rotkehlchen/data_handler.py:59:17:            self.password = new_password
rotkehlchen/data_handler.py:132:13:        self.password = password
rotkehlchen/api/rest.py:1234:53:        if current_password != self.rotkehlchen.data.password:
frontend/app/src/store/session/index.ts:198:40:        description: t('actions.session.password_change.success').toString(),
frontend/app/src/store/session/index.ts:213:40:        description: t('actions.session.password_change.error', {
frontend/app/src/components/account-management/LoginForm.vue:290:32:            :error-messages="v$.password.$errors.map(e => e.$message)"
frontend/app/src/components/account-management/CreateAccountForm.vue:67:54:        'create_account.select_credentials.validation.password_confirmation_mismatch'
frontend/app/src/components/account-management/CreateAccountForm.vue:266:57:                      'create_account.select_credentials.password_sync_requirement'
frontend/app/src/components/account-management/CreateAccountForm.vue:278:38:                  :error-messages="v$.password.$errors.map(e => e.$message)"
frontend/app/src/components/account-management/CreateAccountForm.vue:288:23:                    v$.passwordConfirm.$errors.map(e => e.$message)
frontend/app/src/components/settings/data-security/ChangePassword.vue:36:37:      tc('change_password.validation.password_mismatch'),
frontend/app/src/components/settings/data-security/ChangePassword.vue:87:43:        :label="tc('change_password.labels.password')"
```

So from that you can see `premium_sync` uses it.
`data_handler` **does not**.
`rotkehlchen.py` **does not**.

Only one of those is needed to compare change of password when doing it through rest.py

And then by closer inspection you can see that even then also `premium_sync` does not use it. Since it calls a `data_handler` function which takes a password argument, though data handler already knows the password.

Do you see what I mean by improving code if you touch it?
Do you see what you should do?